yb-voyager/cmd/live_migration.go

```diff
@@ -16,42 +16,65 @@ limitations under the License.
 package cmd
 
 import (
-	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
+	"time"
 
 	log "github.com/sirupsen/logrus"
 
 	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
-	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
 )
 
 func streamChanges() error {
-	queueFilePath := filepath.Join(exportDir, "data", "queue.json")
-	log.Infof("Streaming changes from %s", queueFilePath)
-	file, err := os.OpenFile(queueFilePath, os.O_CREATE, 0640)
-	if err != nil {
-		return fmt.Errorf("error opening file %s: %v", queueFilePath, err)
+	eventQueue := tgtdb.NewEventQueue(exportDir)
+	log.Infof("Streaming changes from %s", eventQueue.QueueDirPath)
+	for { // continuously get next segments to stream
+		segment, err := eventQueue.GetNextSegment()
+		log.Infof("Got next segment to stream: %v", segment)
+		if err != nil {
+			if segment == nil && errors.Is(err, os.ErrNotExist) {
+				log.Info("no segment to stream. Sleeping for 2 second.")
```

Comment:
This line will flood the log.

Reply:
yeah agreed. But is there any other way using which we can tell that there are no segments to read, but less frequently than this